amethyst_tiles/src/pass.rs

```diff
@@ -0,0 +1,503 @@
+    }
+}
+
+fn fetch_world_bounds<T: Tile, E: CoordinateEncoder>(
+    tile_map: &TileMap<T, E>,
+    resources: &Resources,
+) -> (Vector3<u32>, Vector3<u32>) {
+    let camera_fetch = CameraGatherer::gather_camera(resources);
+    assert!(camera_fetch.is_some());
+    let camera_entity = camera_fetch.unwrap();
+
+    let (transforms, cameras) =
+        <(ReadStorage<'_, Transform>, ReadStorage<'_, Camera>)>::fetch(resources);
+
+    //let camera_tile_id = entity_tile_ids.get(camera_entity).u wrap();
+    let camera_transform = transforms.get(camera_entity).unwrap();
+    let camera_translation = camera_transform.global_matrix().column(3).xyz();
+    let camera = cameras.get(camera_entity).unwrap();
+
+    assert!(camera.projection().as_orthographic().is_some());
+    let proj = camera.projection().as_orthographic().unwrap();
+
+    // Scale the camera view area
+    let world_top: Vector3<f32> = amethyst_core::math::convert(
+        camera_translation
+            + Vector3::new(
+                proj.left() * camera_transform.scale().x,
+                proj.top() * camera_transform.scale().y,
+                0.0,
+            ),
+    );
+
+    let world_bottom: Vector3<f32> = amethyst_core::math::convert(
+        camera_translation
+            + Vector3::new(
+                proj.right() * camera_transform.scale().x,
+                proj.bottom() * camera_transform.scale().y,
+                0.0,
+            ),
+    );
+
+    let one = Vector3::new(1, 1, 0);
+
+    let mut world_top_tile = tile_map
+        .to_tile(&world_top)
+        .map_or(Point3::new(0, 0, 0), |p| {
+            if p.x > 1 && p.y > 1 {
+                p - one
+            } else {
+                p
+            }
+        });
+
+    let mut world_bottom_tile =
+        tile_map
+            .to_tile(&world_bottom)
+            .map_or(Point3::from(*tile_map.dimensions()), |mut p| {
+                p += one * 2;
+
+                if p.x > tile_map.dimensions().x {
+                    p.x = tile_map.dimensions().x;
+                }
+                if p.y > tile_map.dimensions().y {
+                    p.y = tile_map.dimensions().y;
+                }
+                if p.z > tile_map.dimensions().z {
+                    p.z = tile_map.dimensions().z;
+                }
+                p
+            });
+
+    world_top_tile.z = 0;
+    world_bottom_tile.z = 0;
+
+    (world_top_tile.coords, world_bottom_tile.coords)
+}
+
+/// This is a placeholder system for setting up the approriate resources for our bundle
+#[derive(Debug, Default)]
+struct SetupSystem<T, E> {
+    _marker: std::marker::PhantomData<(T, E)>,
+}
+impl<'a, T: Tile, E: CoordinateEncoder> System<'a> for SetupSystem<T, E> {
+    type SystemData = (
+        ReadStorage<'a, Handle<SpriteSheet>>,
+        ReadStorage<'a, Handle<Texture>>,
+        ReadStorage<'a, Hidden>,
+        ReadStorage<'a, TileMap<T, E>>,
+    );
+
+    #[cold]
+    fn run(&mut self, _: Self::SystemData) {}
```

Comment:
This is not `cold`. The function does nothing, but is still called every frame. This is not ideal and we should add an option to setup resources directly in the bundle, but this is how it works now.